pkg/k8s/cnp.go

```diff
@@ -268,12 +285,52 @@ retryLoop:
 	return err
 }
 
-func (c *CNPStatusUpdateContext) update(cnp *types.SlimCNP, enforcing, ok bool, cnpError error, rev uint64, cnpAnnotations map[string]string) error {
+// CNPStatusesPath is the prefix in the kvstore which will contain all keys
+// representing CNPStatus state for all nodes in the cluster.
+var CNPStatusesPath = path.Join(kvstore.BaseKeyPrefix, "state", "cnpstatuses", "v2")
+
+func formatKeyForKvstore(uid k8sTypes.UID, namespace, name, nodeName string) string {
+	return path.Join(CNPStatusesPath, string(uid), namespace, name, nodeName)
+}
+
+// UpdateMetadata contains information about how the update for the NodeStatus
+// in a CNP will be propagated.
+type UpdateMetadata struct {
```

Comment:
What are we gaining for having this structure? This seems to be hiding the "context" away of the function arguments. I would suggest to have 2 functions that perform the update, one called "updateToKVStore" and other called "updateToAPIServer". The code duplication of having 2 functions is not big enough that would justify having 2 switch cases in `update`

Reply:
Posted a commit which removes this.